### PR TITLE
fix NPE by adding null check conditional

### PR DESCRIPTION
**Pull request by sumitsum**
## Fix NPE
- Fix NPE seen on Sentry by adding null check for result body: 
  - https://sentry.io/organizations/appsmith/issues/2336226843/?project=1546547&query=is%3Aunresolved 
  - https://sentry.io/organizations/appsmith/issues/2336224002/?project=1546547&query=is%3Aunresolved 

## How Has This Been Tested?
- JUnit TC

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java [hunk 1]
```diff
@@ -710,6 +710,11 @@ private ActionExecutionResult addDataTypes(ActionExecutionResult result) {
             return result;
         }
 
+        if (result.getBody() == null) {
+            result.setDataTypes(List.of(new ParsedDataType(ActionResultDataType.RAW)));
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:714]: Shouldn't this be an empty list? @sumitsum

Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:714]: @nidhi-nair My thinking is that even an null/empty body should show up as an empty text - i.e. the user should be able to see that the return body is empty. @trishaanand need your opinion on this.

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:714]: @trishaanand away right now, but I think if there is no body, the handling on the client side will rely on body being null. This list being empty is an indication of the body being empty as well. The list having any element of type raw could also mean empty body of type raw.

**Review by nidhi-nair — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java]: Done.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**